newIDE/app/src/MainFrame/index.js

```diff
@@ -518,6 +522,29 @@ class MainFrame extends React.Component<Props, State> {
     this.toolbar.setEditorToolbar(editorToolbar);
   };
 
+  setScenePreview = () => {
+    const editorTab = getCurrentTab(this.state.editorTabs);
+    this._setLayoutForPreview(editorTab.name);
+
+    this.setState(
+      {
+        isPreviewOverride: true,
+      },
+      () => this.updateToolbar()
+    );
+  };
+
+  togglePreviewOverride = () => {
+    if (!this.state.previewFirstSceneName) {
+      this.setScenePreview();
+      return;
+    }
```

Comment:
I'm not sure why you would be messing with "previewFirstSceneName" in this function.

This function is "togglePreviewOverride". When called, it must activated or deactivate the boolean that tell if the override is activated. How to do that? Just by doing:

```
    this.setState({ isPreviewOverride: !this.state.isPreviewOverride }, () => {
      this.updateToolbar();
    });
```

Apart from the toolbar that need a manual refresh that's... all you need?

Reply:
With 
```
if (!this.state.previewFirstSceneName) {
      this.setScenePreview();
      return;
    }
```
At start when we open the project and go in SceneA, the checkbox is uncheck and label show "Use this scene for preview".
Then when we check it for the first time the active editor wil be used for preview.
The checkbox is now checked and label show "Use scene SceneA for preview"

If i delete this lines, when i check it and go in another tab like SceneB, the label show "Use this scene for preview", but the override is active on the SceneA, so labels will be wrong if a remove this lines.

This is the reason of this part of code :)

Reply:
This is now deleted, see last commit.